### PR TITLE
Centralize warning configuration

### PR DESCRIPTION
**Pull request by dkalinichenko-js**
Rebase of https://github.com/ocaml-flambda/flambda-backend/pull/1867 on the current main branch.

Clean up warnings in stdlib and otherlibs. Silence some more warnings.

### CODE REVIEW

File: middle_end/flambda2/tests/api_tests/dune [hunk 1]
```diff
@@ -1,10 +1,17 @@
 (tests
  (names extension_meet)
  (modes native)
- (instrumentation (backend bisect_ppx))
- (flags (:standard -principal))
+ (instrumentation
+  (backend bisect_ppx))
```
Comment (lukemaurer) [inline middle_end/flambda2/tests/api_tests/dune:5]: This is one of several files that seems to have been reformatted by an editor. Have we defined a style for dune files somewhere? If not, the reformatting should be left for a different PR.

File: chamelon/dune.jst [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 (env
   (dev
-    (flags (:standard -warn-error -A))))
+    (flags (:standard -no-principal -warn-error -A -w -70)))
+  (_
+    (flags (:standard -no-principal -w -70))))
```
Comment (lukemaurer) [inline chamelon/dune.jst:15]: Is this doing anything when we're turning on `-warn-error` in the top-level `dune` anyway?

File: chamelon/dune.upstream [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
 (env
   (dev
-    (flags (:standard -warn-error -A))))
+    (flags (:standard -no-principal -warn-error -A -w -70)))
+  (_
+    (flags (:standard -no-principal -w -70))))
```
Comment (lukemaurer) [inline chamelon/dune.upstream:15]: Same question as above

**Review by lukemaurer — COMMENTED** [review]
LGTM, besides a couple of minor points and the whitespace changes

**Review by lukemaurer — APPROVED** [review]

**Comment by dkalinichenko-js** [comment]
@Ekdohibs ping, this should be quick :slightly_smiling_face:

**Review by Ekdohibs — APPROVED** [review]
For `chamelon` changes
